FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,942 @@
+  case error
+}
+
+@objc(FIRRemoteConfigError)
+public enum RemoteConfigError: Int, LocalizedError {
+  /// Unknown or no error.
+  case unknown = 8001
+  /// Frequency of fetch requests exceeds throttled limit.
+  case throttled = 8002
+  /// Internal error that covers all internal HTTP errors.
+  case internalError = 8003
+
+  public var errorDescription: String? {
+    switch self {
+    case .unknown:
+      return "Unknown error."
+    case .throttled:
+      return "Frequency of fetch requests exceeds throttled limit."
+    case .internalError:
+      return "Internal error."
+    }
+  }
+}
+
+@objc(FIRRemoteConfigUpdateError)
+public enum RemoteConfigUpdateError: Int, LocalizedError {
+  /// Unable to make a connection to the Remote Config backend.
+  case streamError = 8001
+  /// Unable to fetch the latest version of the config.
+  case notFetched = 8002
+  /// The ConfigUpdate message was unparsable.
+  case messageInvalid = 8003
+  /// The Remote Config real-time config update service is unavailable.
+  case unavailable = 8004
+
+  public var errorDescription: String? {
+    switch self {
+    case .streamError:
+      return "Unable to make a connection to the Remote Config backend."
+    case .notFetched:
+      return "Unable to fetch the latest version of the config."
+    case .messageInvalid:
+      return "The ConfigUpdate message was unparsable."
+    case .unavailable:
+      return "The Remote Config real-time config update service is unavailable."
+    }
+  }
+}
+
+/// Enumerated value that indicates the source of Remote Config data. Data can come from
+/// the Remote Config service, the DefaultConfig that is available when the app is first
+/// installed, or a static initialized value if data is not available from the service or
+/// DefaultConfig.
+@objc(FIRRemoteConfigSource)
+public enum RemoteConfigSource: Int {
+  case remote /// < The data source is the Remote Config service.
+  case `default` /// < The data source is the DefaultConfig defined for this app.
+  case `static` /// < The data doesn't exist, return a static initialized value.
+}
+
+// MARK: - RemoteConfig
+
+private var RCInstances = [String: [String: RemoteConfig]]()
+
+/// Firebase Remote Config class. The class method `remoteConfig()` can be used
+/// to fetch, activate and read config results and set default config results on the default
+/// Remote Config instance.
+@objc(FIRRemoteConfig)
+open class RemoteConfig: NSObject, NSFastEnumeration {
+  /// All the config content.
+  private let configContent: ConfigContent
+
+  private let dbManager: ConfigDBManager
+
+  @objc public var settings: ConfigSettings
+
+  let configFetch: ConfigFetch
+
+  private let configExperiment: ConfigExperiment
+
+  private let configRealtime: ConfigRealtime
+
+  private let queue: DispatchQueue
+
+  // TODO: remove objc public/
+  @objc public let appName: String
+
+  private var listeners = [RemoteConfigListener]()
+
+  public var FIRNamespace: String
```

Comment:
optional: This may be able to be a constant
```suggestion
  public let FIRNamespace: String
```

Reply:
Also should not be public.